audittools/trail.go

```diff
@@ -98,3 +87,22 @@ func (t AuditTrail) Commit(rabbitmqURI, rabbitmqQueueName string) {
 		}
 	}
 }
+
+func (rc *RabbitConnection) refresh(uri, queueName string) {
+	if rc != nil {
+		rc.Disconnect()
+	}
+
+	var err error
+	rc, err = NewRabbitConnection(uri, queueName)
```

Comment:
I may be wrong, but defining a parent pointer inside the pointer's method is kinda dirty. I'd convert this method to a function.

Reply:
Good catch. This is actually not just dirty, it's wrong. Setting `rc` only sets the pointer variable _inside this func_, but it doesn't have an effect on the `rc` held by the caller. This func should return the new RabbitConnection instance so that the caller can use it.

Reply:
My bad, I forgot the `*` 😅 . Fixed in b3fc53dcbb4508f2484348011278a786bdc896c1.

Reply:
Please correct me if I'm wrong but I don't think this is dirty because what we have here is a method with a pointer receiver and the method acts on and changes the *state* of the type that it is defined on.

A use case that comes to mind is implementing `UnmarshalJSON` for a custom type to satisfy the `json.Unmarshaler` interface.

Reply:
`json.Unmarshaler`  updates the pointer value, but not the pointer itself.

Reply:
Ahh, my bad again. I misunderstood.

Are you fine with b3fc53dcbb4508f2484348011278a786bdc896c1?